src/key.rs

```diff
@@ -448,113 +430,37 @@ impl RsaPrivateKey {
-                None,
-                self,
-                ciphertext,
-                &mut *digest,
-                &mut *mgf_digest,
-                label,
-            ),
-            _ => Err(Error::InvalidPaddingScheme),
-        }
+    pub fn decrypt(&self, padding: impl PaddingScheme, ciphertext: &[u8]) -> Result<Vec<u8>> {
+        padding.decrypt(Option::<&mut DummyRng>::None, self, ciphertext)
     }
 
     /// Decrypt the given message.
     ///
     /// Uses `rng` to blind the decryption process.
-    pub fn decrypt_blinded<R: CryptoRngCore>(
+    pub fn decrypt_blinded(
         &self,
-        rng: &mut R,
-        padding: PaddingScheme,
+        rng: &mut impl CryptoRngCore,
+        padding: impl PaddingScheme,
         ciphertext: &[u8],
     ) -> Result<Vec<u8>> {
-        match padding {
-            PaddingScheme::PKCS1v15Encrypt => pkcs1v15::decrypt(Some(rng), self, ciphertext),
-            PaddingScheme::OAEP {
-                mut digest,
-                mut mgf_digest,
-                label,
-            } => oaep::decrypt(
-                Some(rng),
-                self,
-                ciphertext,
-                &mut *digest,
-                &mut *mgf_digest,
-                label,
-            ),
-            _ => Err(Error::InvalidPaddingScheme),
-        }
+        padding.decrypt(Some(rng), self, ciphertext)
     }
 
     /// Sign the given digest.
-    pub fn sign(&self, padding: PaddingScheme, digest_in: &[u8]) -> Result<Vec<u8>> {
-        match padding {
-            // need to pass any Rng as the type arg, so the type checker is happy, it is not actually used for anything
-            PaddingScheme::PKCS1v15Sign { hash_len, prefix } => {
-                if let Some(hash_len) = hash_len {
-                    if digest_in.len() != hash_len {
-                        return Err(Error::InputNotHashed);
-                    }
-                }
-                pkcs1v15::sign::<DummyRng, _>(None, self, prefix.as_ref(), digest_in)
-            }
-            _ => Err(Error::InvalidPaddingScheme),
-        }
+    pub fn sign(&self, padding: impl SignatureScheme, digest_in: &[u8]) -> Result<Vec<u8>> {
+        padding.sign(Option::<&mut DummyRng>::None, self, digest_in)
     }
 
     /// Sign the given digest using the provided rng
     ///
     /// Use `rng` for signature process.
-    pub fn sign_with_rng<R: CryptoRngCore>(
-        &self,
-        rng: &mut R,
-        padding: PaddingScheme,
-        digest_in: &[u8],
-    ) -> Result<Vec<u8>> {
-        match padding {
-            PaddingScheme::PSS {
-                mut digest,
-                salt_len,
-            } => pss::sign::<R, _>(rng, false, self, digest_in, salt_len, &mut *digest),
-            _ => Err(Error::InvalidPaddingScheme),
-        }
-    }
-
-    /// Sign the given digest.
-    ///
-    /// Use `rng` for blinding.
-    pub fn sign_blinded<R: CryptoRngCore>(
+    pub fn sign_with_rng(
```

Comment:
Can you add some clearer docs on when signatures are now blinded?

Reply:
"Blinding" is somewhat conflated between random blinding to mask private key operations and RSA-BSSA Blind Signatures.

Everything related to the latter is now under the `pss` module, namely `Pss:new_blinded` and `Pss::new_blinded_with_salt`, along with `BlindedSigningKey`.

I guess I can spell all of that out in the comment?

Reply:
[f39bf60](https://github.com/RustCrypto/RSA/pull/244/commits/f39bf608b1a0693a285b93396ddaa2087b8154b3) adds more documentation to `sign_with_rng` including information on how to use it with RSA-BSSA blind signatures